src/scripts/components/joinForm.js

```diff
@@ -55,45 +55,106 @@ class JoinForm {
         fields[key].parentNode.classList.remove("giraffe-has-errors")
       );
 
-      if (Email.value.replace(/\D/g, '') === '' && Mobile.value === '' || Name.value === '' || Zip.value === '') {
+      // check for blank required fields
+      if (Email.value === '' && Mobile.value.replace(/\D/g, '') === '' || Name.value === '' || Zip.value === '') {
         e.preventDefault();
         Object.keys(fields).forEach(function(key) { 
           if (fields[key].value === '') {
             fields[key].parentNode.classList.add("giraffe-has-errors");
-            if (key !== "Email") {
-              _this.addErrorToList(key);
+            if (key !== 'Email') {
+              if (key === 'Name' || key === 'Zip') {
```

Comment:
Does fields[key] reliably camel case? If not, is it worth uppercasing both key and the string you're comparing it to?

Reply:
Yes, the keys are case sensitive. It's a tiny aesthetic thing, to have the errors read "Email is required" rather than "email is required".